server/lib/fetch-endpoint.js

```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-const fetch = require('node-fetch');
```

Comment:
Because we switched from `node-fetch` to native `fetch` built in to Node.js, this changed the underlying function doing the request and `@opentelemetry/instrumentation-http` no longer covers us.

We need to make sure the `fetch` is instrumented and update this line:

https://github.com/matrix-org/matrix-public-archive/blob/f3318446f8b71e2e43329ac599ac33f548835169/server/tracing/tracing-middleware.js#L57

Also make sure we serialize it correctly on the timeout page

Reply:
I think we need some instrumentation for `undici`, see https://github.com/open-telemetry/opentelemetry-js-contrib/issues/1021

Reply:
`opentelemetry-instrumentation-fetch-node` seems to work ✅

![Example span from a fetch request using opentelemetry-instrumentation-fetch-node`](https://user-images.githubusercontent.com/558581/235581701-4f79c1f4-c6c3-4de6-9c1a-121af4019abd.png)

I tried `opentelemetry-instrumentation-undici` first but it doesn't seem to add any spans to the traces 🤷